crates/sui-core/src/event_filter.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_core_types::account_address::AccountAddress;
+use move_core_types::identifier::Identifier;
+use move_core_types::language_storage::StructTag;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::BTreeMap;
+use sui_types::base_types::SuiAddress;
+use sui_types::event::{Event, EventEnvelope};
+
+#[derive(Eq, PartialEq, Ord, PartialOrd, Clone, Debug, Serialize, Deserialize)]
+pub enum EventFilter {
+    ByPackage(AccountAddress),
+    ByModule(Identifier),
+    ByFunction(Identifier),
+    ByEventType(StructTag),
```

Comment:
This needs to be the Event type, ie the variant name of the Event enum in `event.rs`.   StructTag is not an event type, it is a definition of the fields for a Move event.   What we are concerned with here is filtering by the type of event, like is it a module Publish event, or a NewObject event, or a MoveEvent.

Reply:
This is for `MoveEventType` will rename it